### ISSUE TITLE
[iOS] [New Architecture 0.73.6] Missing accessibilityLabel Text and Switch Component

### ISSUE THREAD

**Issue by renopp** [facebook/react-native#43648]
### Description

There is an issue with rendering the `Text` and `Switch` components with custom accessibilityLabel on the iOS platform with new architecture enabled. It seems like the accessibilityLabel props not passing properly. However, there is no problem on the Android platform with new architecture enabled.

<img width="756" alt="Screenshot 2024-03-26 at 11 23 05" src="https://github.com/facebook/react-native/assets/11856209/bf9d1bfc-5af5-491b-9672-60f3abc863b8">
<img width="756" alt="Screenshot 2024-03-26 at 11 23 17" src="https://github.com/facebook/react-native/assets/11856209/66845aa6-d715-4cee-9a16-935b2b6d2b23">

### Steps to reproduce

1. `yarn install`
2. `yarn start`
3. `yarn ios`
4. After the app installed successfully open Xcode then go to menu Xcode -> Open Developer Tool -> Accessibility Inspector

### React Native Version

0.73.6

### Affected Platforms

Runtime - iOS

### Areas

Fabric - The New Renderer

### Output of `npx react-native info`

```text
System:
  OS: macOS 13.5.2
  CPU: (8) arm64 Apple M1 Pro
  Memory: 84.02 MB / 16.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.10.0
    path: /usr/local/bin/node
  Yarn:
    version: 1.22.19
    path: ~/.yarn/bin/yarn
  npm:
    version: 8.12.1
    path: /opt/homebrew/bin/npm
  Watchman:
    version: 2023.12.04.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.12.1
    path: /usr/local/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.2
      - iOS 17.2
      - macOS 14.2
      - tvOS 17.2
      - visionOS 1.0
      - watchOS 10.2
  Android SDK: Not Found
IDEs:
  Android Studio: 2023.1 AI-231.9392.1.2311.11255304
  Xcode:
    version: 15.2/15C500b
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.9
    path: /usr/bin/javac
  Ruby:
    version: 2.6.10
    path: /usr/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.73.6
    wanted: 0.73.6
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: true
  newArchEnabled: true
```

### Stacktrace or Logs

```text
no stacktrace
```

### Reproducer

https://github.com/renopp/accessibilityLabel-issue

**Comment by gedeagas** [comment]
Hello @renopp ,

Thank you for reporting this issue. Indeed, it appears to be a bug, and currently, there doesn't seem to be a quick workaround.

With the new architecture in iOS, the `testID` is now being mapped to the `accessibilityIdentifier` of a native component. Previously, it was mapped to the `accessibilityLabel`. This change has led to a situation where **automated testing tools like Appium are unable to find elements using `accessibilityLabel`.**

At the moment, it seems the only viable solution for iOS is to refactor your automated tests to use `accessibilityIdentifier`.

For related issues, please see: https://github.com/facebook/react-native/issues/38709

Unfortunately, I don't believe the React Native core team has a clear timeline for addressing this issue right now.

**Comment by cortinico** [comment]
cc @j-piasecki

**Comment by j-piasecki** [comment]
@renopp Looks like setting `accessibilityLabel` on `Switch` components works correctly now (tested on 3cb03cb2973f36976c95e4ae70c9edc779546ad0):
<img width="1509" alt="Screenshot 2024-06-14 at 16 13 27" src="https://github.com/facebook/react-native/assets/21055725/2e7d692d-e709-437b-a6be-5b9daac59651">

As for the label being wrong for the text component, the prop is being passed properly but it's being overridden by the text value: https://github.com/facebook/react-native/blob/7061649c03d6a121cd5d2bd77daf60166d9b3188/packages/react-native/React/Fabric/Mounting/ComponentViews/Text/RCTParagraphComponentView.mm#L157-L160

The override was added here: https://github.com/facebook/react-native/commit/76a1db4b203f079ca55ded9b093fc5556c5a173c

Would doing something like this:
```objc
- (NSString *)accessibilityLabel
{
  NSString *label = super.accessibilityLabel;
  if ([label length] > 0) {
    return label;
  }
  return self.attributedText.string;
}
```
be acceptable? On the first glance it should handle both cases cc @cortinico @sammy-SC

**Comment by cortinico** [comment]
Thanks for checking this @j-piasecki super valuable!

> Would doing something like this:

Yup your suggested snipped sounds reasonable to me 👍 Don't you mind sending it out as a PR?

**Comment by j-piasecki** [comment]
Here it is: https://github.com/facebook/react-native/pull/45003
